src/de.rs

```diff
@@ -194,6 +240,9 @@ enum ErrorKind {
     /// Unquoted string was found when quoted one was expected
     UnquotedString,
 
+    /// An IO error occurred.
+    Io,
```

Comment:
I would prefer to not add a new error variant here because toml parsing in general doesn't have I/O errors, that's a higher-level concern. I'd personally prefer to add a new error type which is basically either an I/O error or a toml parsing error.